### DISCUSSION TITLE
Deduplication for `husky.add()`

### DISCUSSION THREAD

**Discussion by SevenOutman** [typicode/husky#1012] [Ideas]
It's a common use case that we use `husky` API to build something like a `mrm` task, e.g. `mrm-task-lint-staged`. The script adds `npx lint-staged` into `pre-commit` hook using `husky.add()` API. But if the script was run multiple times, there would also be multiple duplicated lines of `npx lint-staged` inside `pre-commit` script. I believe it would be really helpful if `husky.add()` supports deduplication.

## Expected behavior

1. Avoid duplication in form of exact same line. For example, given `.husky/pre-commit` script as shown below:

```
#!/bin/sh
. "$(dirname "$0")/_/husky.sh"

npx lint-staged
```

Calling `husky.add('.husky/pre-commit', 'npx lint-staged')` should not take effect because there is already a line with exact same content.

2. Furthermore, avoid duplication in a sequence of commands. For example, given `.husky/pre-commit` script as shown below:

```
#!/bin/sh
. "$(dirname "$0")/_/husky.sh"

echo 'Haha' && npx lint-staged
```

Calling `husky.add('.husky/pre-commit', 'npx lint-staged')` should not take effect because there is already a line with the same command called in a sequence.

**Comment by typicode** [comment]
There may be some cases where people would really want to have two times the same command:

```sh
date
# some long command
date
```

I prefer to keep husky simple and not too smart about it. There's `husky set ...` (husky.set()) on the other side which completely replaces the hook.

If checking for duplication is really needed, I'd use an `if` and some regexp maybe before calling `husky.add` :)

**Comment by SevenOutman** [comment]
> If checking for duplication is really needed, I'd use an if and some regexp maybe before calling husky.add :)

Then what about something like a `husky.exists` method? That way users don't have to read/write husky files with a single fs API.
